form-editor-studio-plugin/src/main/joo/com/tallence/formeditor/studio/helper/FormElementsManager.as

```diff
@@ -145,5 +102,68 @@ public class FormElementsManager {
     return dragActiveVE;
   }
 
+  private function getRootNodeStruct():Struct {
+    if (formElements) {
+      return formElements.getValueAsStruct();
+    } else {
+      // if the sub struct is missing the sub struct has to be created
+      var root:Struct = (store.getRoot() as StructTreeNode).getValueAsStruct();
+      root.getType().addStructProperty(FormElementStructWrapper.FORM_ELEMENTS_PROPERTY);
+      return root.get(FormElementStructWrapper.FORM_ELEMENTS_PROPERTY);
+    }
+  }
+
+  /**
+   * Initializes the struct tree store. The store can be used to handle added or removed form elements, so that the
+   * value expression evaluating to the list of applied form elements can be updated. A functional value expression can
+   * not be used to evaluate the form elements directly, because the value expression would be evaluated too often.
+   * Every time a property of a form element is changed, for example the name of the field, the value expression would
+   * recalculate all form elements and the ui would be re-rendered.
+   */
+  private function initTreeStore():void {
+    var storeConfig:StructTreeStore = StructTreeStore({});
+    storeConfig.bindTo = contentVE;
+    storeConfig.propertyName = formDataStructPropertyName;
+    store = new StructTreeStore(storeConfig);
+    store.addListener("nodeappend", nodeAppended);
+    store.addListener("nodeinsert", nodeAppended);
+    store.addListener("noderemove", nodeRemoved);
+  }
+
+  private function nodeRemoved(store:StructTreeStore, record:StructTreeNode):void {
+    var depth:Number = record.getDepth();
+    if (depth == 1) {
+      // If the root node is removed another document form without a 'formElements' sub struct is opened. Therefore
+      // the value expresison has to be set to an empty array
+      formElements = null;
```

Comment:
Es ist auf Anhieb schwer verständlich, was der Unterschied zwischen "formElements" und "formElementsVE" ist. Letzteres scheint eine Liste von Wrappern für die FormElemente zu sein, ersteres eine Struct-Repräsentation für das gesamte Struct. 
Was hälst du von neuen Namen?
formElements -> formElementsStruct
formElementsVE -> formElementWrappersVE

Und warum rufst du in Zeile 139 direkt "setValue()" auf die "getFormElementsVE" auf? Das sollte m.E. auch über der Methode "updateFormElements()" passieren

Reply:
erledigt.